apps/bors_worker/test/batcher/batcher_test.exs

```diff
@@ -438,6 +493,92 @@ defmodule BorsNG.Worker.BatcherTest do
       }}
   end
 
+  test "full with differing branches", %{proj: proj} do
+    # Projects are created with a "waiting" state
+    GitHub.ServerMock.put_state(%{
+      {{:installation, 91}, 14} => %{
+        branches: %{"master" => "ini", "staging" => "", "staging.tmp" => ""},
+        comments: %{1 => [], 2 => []},
+        statuses: %{},
+        files: %{"staging" => %{"bors.toml" => ~s/status = [ "ci" ]/}}
+      }})
+    patch = %Patch{
+      project_id: proj.id,
+      pr_xref: 1,
+      commit: "N",
```

Comment:
I read your test case like 3 times thinking so if release has 0 commits and master has N... 
(facepalm)